updater/lib/dependabot/dependency_change.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+# This class describes a change to the project's Dependencies which has been
+# determined by a Dependabot operation.
+#
+# It includes a list of changed Dependabot::Dependency objects, an array of
+# Dependabot::DependencyFile objects which contain the changes to be applied
+# along with any Dependabot::GroupRule that was used to generate the change.
+#
+# This class provides methods for presenting the change set which can be used
+# by adapters to create a Pull Request, apply the changes on disk, etc.
+module Dependabot
+  class DependencyChange
```

Comment:
I'm really digging this extraction! 👏 on noticing the need and finding a sane way to address it 🙇 

I can tell you've spent time thinking about what this object should include/exclude. I find myself wondering whether `job` is part of a dependency change. My thinking is that a job might _have a_ dependency change, while the rest of the parameters are _parts of_ a dependency change. 🤔

I see `job` is only referenced in `#pr_message`. Given that, we _could_ let it be an argument to that method. I kinda like that: `dependency_change.pr_message(for: job)`

I'll trust your judgment here. I think this is a big improvement, and points us in a really positive direction 🥇

Reply:
Yeah, I'm 50/50 on this class knowing about the Job - I think of it as "I know the context in which this change was made" as the `base_commit_sha` could conceivably be retrieved from it as well and the `Updater#existing_pull_request` could sink into `DependencyChange` as well if the `Job` maintained the list of Pull Requests created so far.

Essentially I think this feels untidy as it is an unfinished piece; I think this class should know about the 'context' in which it is created, the job is _part_ of that context and good enough to use for now.

The thing I'm not sure about is, should the Job _be_ the context or should the Job and some other things be split out into a new class representing just context?